src/feature/nodelist/microdesc.c

```diff
@@ -221,6 +221,13 @@ dump_microdescriptor(int fd, microdesc_t *md, size_t *annotation_len_out)
   }
 
   md->off = tor_fd_getpos(fd);
+  const char *nulpos = memchr(md->body, 0, md->bodylen);
+  if (BUG(nulpos)) {
+    log_warn(LD_BUG, "About to dump a NUL into a microdescriptor file. "
+             "offset %"PRId64", bodylen %zu, nul position %zu",
+             (int64_t)md->off, md->bodylen,
+             (size_t)(nulpos - md->body));
```

Comment:
Can we log the escaped() context of the NUL?
Maybe 10 bytes either side of the NUL?

Reply:
Also, can we log some identifier from the microdescriptor?

Reply:
Should we call warn_if_nul_found() here?

Reply:
I'll try to do the other stuff; at this stage, there is no identifier from the md that we could log.